### ISSUE TITLE
Dashboard: Cannot complete a truncated item (a long-standing issue remains)

### ISSUE THREAD

**Issue by dwertheimer** [NotePlan/plugins#593]
Content in long paragraph is being truncated and therefore the matching paragraph cannot be found:
```
2024-09-10 14:30:07 | DEBUG | *************** bridgeClickDashboardItem: completeTask ***************
2024-09-10 14:30:07 | DEBUG | NPP/completeItem :: starting with filename: 20240827.md, content: "Buy tickets for the mint [Emma Garfin, Velo Li, Liana Jaden, Ella Clayton Tickets | From $13.39 | Sep 18 @ The Mint , Los Angeles | DICE](https://dice.fm/event/k662v9-emma-garfin-velo-li-liana-jaden-ella-clayton-18th-sep-the-mint-los-angeles-tickets?_branch_referrer=H4sIAAAAAAAAA8soKSkottLXz8nMy9ZLy..."
2024-09-10 14:30:07 🥺 WARN🥺 NPP/findParaFromStringAndFilename :: Couldn't find paragraph {Buy tickets for the mint [Emma Garfin, Velo Li, Liana Jaden, Ella Clayton Tickets | From $13.39 | Sep 18 @ The Mint , Los Angeles | DICE](https://dice.fm/event/k662v9-emma-garfin-velo-li-liana-jaden-ella-clayton-18th-sep-the-mint-los-angeles-tickets?_branch_referrer=H4sIAAAAAAAAA8soKSkottLXz8nMy9ZLy...} in note '20240827.md'
```

**Comment by jgclark** [comment]
Probably a dupe of #587.
This was closed in (I think) 2.1.0.a11.

**Comment by dwertheimer** [comment]
@jgclark I am not sure why this keeps getting closed. Does it work for you? The examples I provide in both issues still fail for me. And today I had another one. The following task was a very long share sheet task I generated from a video/article I saw:

```
* Stretch: [Dr. Joe Damiani - TMJ, Head & Neck Specialist on Instagram: "Comment the Word ‘Headaches’ on This Post if You Need Help Getting Rid of Them, and I’ll Help You Out!Do You Experience Shoulder Blade Pain That Just Won’t Go Away?  You’ve Tried Stretching It, Massaging It and Fixing Your Posture but It Keeps Coming Back?  Plus You Get Headaches and It Seems Like It’s All Connected? (Well You’re Probably Right!)You See When We Have Improper Shoulder Blade Function, the Muscles That Connect It to the Spine Have to Overwork and Those Muscles Tug on the Neck Which Can Cause Headaches!PLUS, if Your Upper Back and Rib Head Joint Are Stiff… Then the Muscles (Rhomboids, Levant Scapulae and Upper Trapezius Have to Work Even More). SO to Fix the Situation We Have To:1.) Release the Muscles to Put Them in a More Healthy State.2.) Mobilize the Upper Spine and Rib Head Joints3.) Retrain the Serrates Anterior to Properly Move the Shoulder Blade So We Don’t Have Any Further Compensations by the Headache Causing Muscles!I Fine It So Incredibly Common That When Your Neck and Shoulders Do Not Function Properly, They Are Secretly Causing Your Headaches!  Try This Out and if You Need More Help or to Figure Out Any Other Root Cause, Comment ‘HEADACHE’ on This Post and I’ll Help You Out!#Headaches #Headacherelief #Neckpain #Neckpainrelief #Migraine #Tmj #Tmjdisorder #Tmd"](https://www.instagram.com/reel/C_jT05UxAIR/?igsh=MTc4MmM1YmI2Ng==) 
```
When I try to check this as done in the dashboard Task Dialog, the logs say this:
```
[WebView Log] 2024-09-26 13:48:03 | DEBUG | convertBoldAndItalicToHTML :: -> Stretch: [Dr. Joe Damiani - TMJ, Head & Neck Specialist on Instagram: "Comment the Word ‘Headaches’ on This Post if You Need Help Getting Rid of Them, and I’ll Help You Out!Do You Experience Shoulder Blade Pain That Just Won’t Go Away?  You’ve Tried Stretching It, Massaging It and Fixing Your Postur...

...

2024-09-26 13:48:03 | DEBUG | ***************** bridgeClickDashboardItem: showLineInEditorFromFilename *****************
Note opened notification, filename = 20240912.md, waiting for: 20240912.md, matches = true
2024-09-26 13:48:04 | DEBUG | highlightParagraphInEditor :: Looking for <Stretch: [Dr. Joe Damiani - TMJ, Head & Neck Specialist on Instagram: "Comment the Word ‘Headaches’ on This Post if You Need Help Getting Rid of Them, and I’ll Help You Out!Do You Experience Shoulder Blade Pain That Just Won’t Go Away?  You’ve Tried Stretching It, Massaging It and Fixing Your Postur...>
2024-09-26 13:48:04 | DEBUG | NPParagraph :: findParagraph: found no paragraphs in note "20240912.md" that matches undefined
2024-09-26 13:48:04 🥺 WARN🥺 highlightParagraphInEditor :: Sorry, couldn't find paragraph with rawContent <undefined> to highlight in open note
2024-09-26 13:48:04 | DEBUG | bridgeClickDashboardItem :: -> successful call to open filename 20240912.md in Editor, followed by unsuccessful call to highlight the paragraph in the editor
```

If it gives you any clues, when I log the object when it's being rendered on the React side:
```
[WebView Log] 2024-09-26 14:15:27 | DEBUG | ItemContent :: {
  "item": {
    "ID": "13-1",
    "itemType": "open",
    "para": {
      "filename": "20240912.md",
      "noteType": "Calendar",
      "title": "2024-09-12",
      "type": "open",
      "prefix": "* ",
      "content": "Stretch: [Dr. Joe Damiani - TMJ, Head & Neck Specialist on Instagram: \"Comment the Word ‘Headaches’ on This Post if You Need Help Getting Rid of Them, and I’ll Help You Out!Do You Experience Shoulder Blade Pain That Just Won’t Go Away?  You’ve Tried Stretching It, Massaging It and Fixing Your Postur...",
      "rawContent": "* Stretch: [Dr. Joe Damiani - TMJ, Head & Neck Specialist on Instagram: \"Comment the Word ‘Headaches’ on This Post if You Need Help Getting Rid of Them, and I’ll Help You Out!Do You Experience Shoulder Blade Pain That Just Won’t Go Away?  You’ve Tried Stretching It, Massaging It and Fixing Your Postur...",
      "priority": -1,
      "timeStr": "none",
      "startTime": "none",
      "changedDate": "2024-09-26T20:47:17.198Z",
      "hasChild": false,
      "isAChild": false,
      "indentLevel": 0
    }
  },
  "actionType": "(not yet set)",
  "sectionCodes": [
    "OVERDUE"
  ]
}
```

Looks like both content and rawContent are are truncated. So I assume this is being done in the dataGeneration but I don't know that code well enough to be sure. Hope that helps a little.

**Comment by jgclark** [comment]
Yes, I got it working with your earlier examples.
I'll look again with your latest example.

**Comment by dwertheimer** [comment]
@jgclark having this on many many items now on the released 2.0.6:
I tend to include links to gmails a lot. here is an example:
```
* ! Call usaa 800-531-0014 Option 2 re umbrella claim [email](https://mail.google.com/mail/u/0/popout?ver=1we8350igu8z&q=usaa%20agency&qs=true&qid=0E04B4CF-E9BF-4031-BCBB-2B22A462271B&aqid=6AA969D5-EC70-491A-9706-D81C0D0ED0B9&search=query&silk=B1D1F891-9AD2-4877-9BB7-8BB624F3CBD2&th=%23thread-f%3A1813015901883187288&qt=agencies.1.agency.1.agency%27s.1.agent.1.usaa.1.usaa%27s.1&cvid=1) rental reimbursement coverage is still open - Kenna claim dated August 6, 2024 - why is kenna claim included?
```
Log:
```
2024-10-21 14:54:51 | DEBUG | NPP/completeItem :: starting with filename: 20241016.md, content: "! Call usaa 800-531-0014 Option 2 re umbrella claim [email](https://mail.google.com/mail/u/0/popout?ver=1we8350igu8z&q=usaa%20agency&qs=true&qid=0E04B4CF-E9BF-4031-BCBB-2B22A462271B&aqid=6AA969D5-EC70-491A-9706-D81C0D0ED0B9&search=query&silk=B1D1F891-9AD2-4877-9BB7-8BB624F3CBD2&th=%23thread-f%3A1813..."
2024-10-21 14:54:51 🥺 WARN🥺 NPP/findParaFromStringAndFilename :: Couldn't find paragraph {! Call usaa 800-531-0014 Option 2 re umbrella claim [email](https://mail.google.com/mail/u/0/popout?ver=1we8350igu8z&q=usaa%20agency&qs=true&qid=0E04B4CF-E9BF-4031-BCBB-2B22A462271B&aqid=6AA969D5-EC70-491A-9706-D81C0D0ED0B9&search=query&silk=B1D1F891-9AD2-4877-9BB7-8BB624F3CBD2&th=%23thread-f%3A1813...} in note '20241016.md'
```

**Comment by jgclark** [comment]
That content line works for me in 2.1.0.a13, which is the latest I have built.  I can't currently build 2.0.7, which I thought until now was ready to release to go with the updated Reviews plugin. I can't solve that now.

In 2.0.6 I've chased the chain of calls NPP/findPara... <<- completeItem(). That is called by doCompleteTask which has first called validateAndFlattenMessageObject to get the `content` parameter. It's that that appears to be truncated when you don't want it to be. You're going to need to look at that function, as I don't fully understand it.
